llama_index/core/base_multi_modal_retriever.py

```diff
@@ -13,15 +13,27 @@ class MultiModalRetriever(BaseRetriever, BaseImageRetriever):
 
     @abstractmethod
     def text_retrieve(self, str_or_query_bundle: QueryType) -> List[NodeWithScore]:
-        """Retrieve text nodes given query.
+        """Retrieve text nodes given text query.
 
         Implemented by the user.
 
         """
 
     @abstractmethod
-    def image_retrieve(self, str_or_query_bundle: QueryType) -> List[NodeWithScore]:
-        """Retrieve image nodes given query.
+    def text_to_image_retrieve(
+        self, str_or_query_bundle: QueryType
+    ) -> List[NodeWithScore]:
+        """Retrieve image nodes given text query.
+
+        Implemented by the user.
+
+        """
+
+    @abstractmethod
+    def image_to_image_retrieve(
+        self, str_or_query_bundle: QueryType
```

Comment:
input type seems wrong

Reply:
oh it is as expected. query_bundle now stores image_path and we could return image_path by calling `embedding_image`